src/Oscoin/Storage/ContentHashStore.hs

```diff
@@ -0,0 +1,73 @@
+-- | 'ContentHashStore' is an interface for storing values adressed by
+-- their hashes. Conceptually 'ContentHashStore' is a mutable 'Map'
+-- with hashes as keys.
+module Oscoin.Storage.ContentHashStore
+    ( ContentHashStore
+    , storeHashContent
+    , lookupHashContent
+
+    , newContentHashStoreIO
+    , mkStateContentHashStore
+    , hoistContentHashStore
+    ) where
+
+import           Oscoin.Prelude
+
+import qualified Oscoin.Crypto.Hash as Crypto
+
+import qualified Data.Map.Strict as Map
+import           Lens.Micro
+import           Lens.Micro.Mtl
+
+data ContentHashStore crypto value m = ContentHashStore
+    { _storeHashContent  :: value -> m ()
+    , _lookupHashContent :: Crypto.Hashed crypto value -> m (Maybe value)
+    }
```

Comment:
By convention, we use `Handle` for this and then we would use it externally as `ContentHashStore.Handle`

Reply:
I assumed we only use `Handle` for the old MTL-style APIs, like `StateStore.Handle`. The new dictionary APIs, like `BlockStore` does not use the term `Handle`. Also this module is intended to be imported unqualified as all the names include `ContentHash`. If we want to keep it at that we would need to use the name `ContentHashStoreHandle` which would be quite long. `ContentHashStore` flows better I think.

Reply:
Hmm yeah I think that makes sense, since the Handle is not actually opaque here, as it is with mempool, telemtry etc.